packages/segmented-control/src/SegmentedControlOption.tsx

```diff
@@ -0,0 +1,238 @@
+    --text-transform: uppercase;
+    --font-weight: bold;
+    --divider-height: 12px;
+  `}
+  ${size === 'default' &&
+  css`
+    --font-size: 14px;
+    --line-height: 24px;
+    --inline-padding: 3px;
+    --text-transform: none;
+    --font-weight: normal;
+    --divider-height: 18px;
+  `}
+  ${size === 'large' &&
+  css`
+    --font-size: 16px;
+    --line-height: 28px;
+    --inline-padding: 4px;
+    --text-transform: none;
+    --font-weight: normal;
+    --divider-height: 20px;
+  `}
+
+  ${mode === 'light' &&
+  css`
+    --base-text-color: ${uiColors.gray.dark1};
+    --base-background-color: transparent;
+    --base-shadow-color: transparent;
+    --hover-text-color: ${uiColors.gray.dark3};
+    --hover-background-color: ${uiColors.white};
+    --active-text-color: ${uiColors.gray.dark3};
+    --disabled-text-color: ${uiColors.gray.light1};
+  `}
+  ${mode === 'dark' &&
+  css`
+    --base-text-color: ${uiColors.gray.light1};
+    --base-background-color: transparent;
+    --base-shadow-color: transparent;
+    --hover-text-color: ${uiColors.gray.light2};
+    --hover-background-color: ${uiColors.gray.dark2};
+    --active-text-color: ${uiColors.white};
+    --disabled-text-color: ${uiColors.gray.dark1};
+  `}
+
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: var(--inline-padding) 12px;
+  border-radius: 4px;
+  text-align: center;
+  font-size: var(--font-size);
+  line-height: var(--line-height);
+  text-transform: var(--text-transform, none);
+  font-weight: var(--font-weight);
+  color: var(--base-text-color); // color
+  background-color: var(--base-background-color); // color
+  box-shadow: 0px 1px 2px var(--base-shadow-color); // color
+  cursor: pointer;
+  transition: all 100ms ease-in-out;
+  z-index: 1;
+
+  &:hover {
+    color: var(--hover-text-color);
+
+    &:after {
+      background-color: var(--hover-background-color);
+    }
+
+    &${checkedSelector}, &[data-disabled='true'] {
+      border-color: transparent;
+      background-color: transparent;
+
+      &:after {
+        background-color: transparent;
+      }
+    }
+  }
+
+  &${checkedSelector} {
+    color: var(--active-text-color); // color
+  }
+
+  &[data-disabled='true'] {
+    color: var(--disabled-text-color); // color
+    cursor: not-allowed;
+  }
+
+  // Hover indicator
+  &:after {
+    content: '';
+    position: absolute;
+    height: calc(100% - 2px);
+    width: 100%;
+    top: 1px;
+    left: 0;
+    background-color: transparent;
+    z-index: -1;
+    border-radius: inherit;
+    transition: all 100ms ease-in-out;
+  }
+
+  /* 
+   * Adds the divider line to unselected segments 
+   */
+  &:before {
+    content: '';
+    position: absolute;
+    height: var(--divider-height);
+    width: 1px;
+    left: calc(0px - (var(--segment-gap, 1px) + 1px) / 2);
+    background-color: transparent;
+    transition: all 100ms ease-in-out;
+  }
+
+  &${uncheckedSelector} {
```

Comment:
I think this selector needs to be rewritten. Right now, the output selector would be:

```css
.classname[data-checked="false"]:not(:first-child):not([data-checked="true"] + [data-checked="false"]):before
```

Assuming that the `:not([data-checked="true"] + [data-checked="false"])` portion also will have the same class name, you could write something like this instead:

```css
.classname ~ .classname[data-checked="false"]:before {
  background-color: ${uiColors.gray.light1};
}

.classname[data-checked="true"] + [data-checked="false"]:before {
  background-color: unset;
}

```


We can potentially simplify even further if you have the border added to the right of the element, rather than to the left.

```css
.classname[data-checked="false"]:after {
	background-color: ${uiColors.gray.light1};
}

.classname[data-checked="true"],
.classname:last-of-type {
	&:after {
    background-color: unset;
	}
}
```

Disclaimer: This is going from what I'm reading on this first pass – I haven't tested this locally yet.